core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/controller/Controller.scala

```diff
@@ -132,6 +155,60 @@ class Controller(
       this.handleControlPayloadWithTryCatch(CLIENT, invocation)
   }
 
+  def acceptRecoveryMessages: Receive = {
+    case recoveryMsg: WorkflowRecoveryMessage =>
+      recoveryMsg.payload match {
+        case UpdateRecoveryStatus(isRecovering) =>
+          logger.info("recovery status for " + recoveryMsg.from + " is " + isRecovering)
+          globalRecoveryManager.markRecoveryStatus(recoveryMsg.from, isRecovering)
+        case ResendOutputTo(vid, ref) =>
+          logger.warn(s"controller should not resend output to " + vid)
```

Comment:
add another message to let controller immediately resend unacked control messages to worker

Reply:
will do it in another PR.